### PR TITLE
INT-3589: Upgrade dependencies

### PR DESCRIPTION
**Pull request by artembilan**
JIRA: https://jira.spring.io/browse/INT-3589, https://jira.spring.io/browse/INT-3624
- SF - 4.2. Fix Breaking changes for the `ApplicationEventPublisher`
- AMQP 1.5. Without issues
- Reactor - 2.0. Fix for the new `Stream` foundation
- jsonPath - 1.2.0. Fix for new `Predicate` abstraction. Add 'fail-fast error' to the `IntegrationRegistrar`
- Sshd - 0.13.0. Fix for new `VirtualFileSystemFactory` usage
- Spring Data - Fowler
- And others without issues
- Get rid of `reactor.util.StringUtils` usage

### CODE REVIEW

File: spring-integration-core/src/main/java/org/springframework/integration/config/IntegrationRegistrar.java [hunk 1]
```diff
@@ -220,6 +220,17 @@ private void registerBuiltInBeans(BeanDefinitionRegistry registry) {
 						"there is no jayway json-path.jar on the classpath.");
 			}
 
+			try {
+				ClassUtils.forName("com.jayway.jsonpath.Predicate", this.classLoader);
+			}
+			catch (ClassNotFoundException e) {
+				jsonPathClass = null;
+				logger.error("SpEL function '#jsonPath' isn't registered. " +
```
Comment (garyrussell) [inline spring-integration-core/src/main/java/org/springframework/integration/config/IntegrationRegistrar.java]: I think we should only log this as an error if `JsonPath` __is__ on the CP - they might not be using json path at all.

Comment (artembilan) [inline spring-integration-core/src/main/java/org/springframework/integration/config/IntegrationRegistrar.java]: Ah! Good catch! And not `error`, but `warn`

Comment (garyrussell) [inline spring-integration-core/src/main/java/org/springframework/integration/config/IntegrationRegistrar.java]: No; I mean if we've already detected that json path is __not__ on the CP we shouldn't even do this test. It should be in the 

    if (jsonPathClass != null) {
        ...
    }

below

Comment (artembilan) [inline spring-integration-core/src/main/java/org/springframework/integration/config/IntegrationRegistrar.java]: Yes, I got it. But the "they might not be using json path at all" argument is against of `error`

File: spring-integration-core/src/test/java/org/springframework/integration/aggregator/AggregatorTests.java [hunk 2]
```diff
@@ -76,6 +76,12 @@ public void configureAggregator() {
 			public void publishEvent(ApplicationEvent event) {
 				expiryEvents.add((MessageGroupExpiredEvent) event);
 			}
```
Comment (garyrussell) [inline spring-integration-core/src/test/java/org/springframework/integration/aggregator/AggregatorTests.java]: 2015

Comment (artembilan) [inline spring-integration-core/src/test/java/org/springframework/integration/aggregator/AggregatorTests.java]: Done. Wait for more feedback to push

File: spring-integration-core/src/test/java/org/springframework/integration/channel/QueueChannelTests.java [hunk 3]
```diff
@@ -248,6 +249,7 @@ public void testPurge() {
 	public final TemporaryFolder tempFolder = new TemporaryFolder();
 
 	@Test
+	@Ignore
```
Comment (garyrussell) [inline spring-integration-core/src/test/java/org/springframework/integration/channel/QueueChannelTests.java]: Why is this ignored? It sometimes works for me; sometimes fails.

And, there is a bad `latch2.await();` (no timeout) (and latch3)

If this is a known reactor problem, add a comment pointing to the reactor issue after `@Ignore`

Comment (artembilan) [inline spring-integration-core/src/test/java/org/springframework/integration/channel/QueueChannelTests.java]: Ah! Yes, forgot to do that: https://jira.spring.io/browse/INT-3644

**Comment by garyrussell** [comment]
That's all for me; after you push and I merge I'll rebase and fix the new TCP event test in my PR.

**Comment by artembilan** [comment]
Pushed

**Comment by garyrussell** [comment]
Merged as f248394682b4f5165e4d1fbc40a1b66d2e9996ff Polish here:https://github.com/garyrussell/spring-integration/commit/d6655a5723ee123a79eddc6943a160bc21e72104

**Pull request closed without merging**